recipe/build-mpi.sh

```diff
@@ -10,6 +10,11 @@ export FC=$(basename "$FC")
 # unset unused Fortran compiler variables
 unset FFLAGS F77 F90 F95
 
+# add -fallow-argument-mismatch to FCFLAGS
+if [[ "$target_platform" != linux-ppc64le ]]; then
+   export FCFLAGS="-fallow-argument-mismatch ${FCFLAGS}"
+fi
```

Comment:
The `export FCFLAGS="-fallow-argument-mismatch"` flag is used to prevent the GNU Fortran compiler (starting from version 10) from generating errors for argument mismatches in function and subroutine calls. This ensures compatibility with legacy Fortran code or codebases that do not strictly follow argument type matching, allowing for smoother compilation without interruptions.

Reply:
Thanks, @j34ni. A follow-up question: This flag is only used at build time, but Open MPI was already built fine, so what issues does this solve? I don't think I get it.

Reply:
This is because if it is not here there can be arguments mismatches later in codes compiled with mpifort (that happens with CESM typically), so I added it to prevent this kind of errors

Reply:
Does the build-time `FCFLAGS` get recorded in `mpifort`? If not, I would think this is added to the wrong place, and it should be added to the activation script of `openmpi-mpifort` instead. Furthermore, by doing so it would cause a global change in whoever using `openmpi-mpifort`, and I would be hesitant in propagating to downstream projects/users. @minrk @dalcinl thoughts?

Reply:
CFLAGS/etc do not get recorded in the wrappers, that would be in `--wrapper-cflags` args, as we currently have for `--wrapper-ldflags` ([docs](https://docs.open-mpi.org/en/v5.0.x/installing-open-mpi/compilers-and-flags.html)). This arg here only affects the build of openmpi itself, so if openmpi itself doesn't need it, we can leave it out.